index.js

```diff
@@ -67,11 +67,11 @@ function rateLimitPlugin (fastify, settings, next) {
   }
 
   // onRoute add the onRequest rate-limit function if needed
-  fastify.addHook('onRoute', (routeOptions) => {
+  fastify.addHook('onRoute', async (routeOptions) => {
```

Comment:
Unfortunately using `async` here is not supported.

Reply:
I'll fix this.